### PR TITLE
Remove String Jaccard Index

### LINKED ISSUE #55078: `byteJaccardIndex` is slow
**Describe the unexpected behaviour**

```
        std::unordered_set<char> haystack_set(haystack, haystack + haystack_size);
        std::unordered_set<char> needle_set(needle, needle + needle_size);
```

```
play-eu :) SELECT DISTINCT arrayJoin(tokens(lower(text))) AS word, byteJaccardIndex(word, 'clickhouse') AS distance FROM hackernews ORDER BY distance DESC LIMIT 20

SELECT DISTINCT
    arrayJoin(tokens(lower(text))) AS word,
    byteJaccardIndex(word, 'clickhouse') AS distance
FROM hackernews
ORDER BY distance DESC
LIMIT 20

Query id: 5d3d9a9a-eb45-42a5-92d2-d38738dc32ea

┌─word────────────┬───────────distance─┐
│ clickhouse      │                  1 │
│ clickhouses     │                  1 │
│ clikhouse       │                  1 │
│ clickehouse     │                  1 │
│ clikchouse      │                  1 │
│ chickenlicious  │                0.9 │
│ bokehlicious    │                0.9 │
│ 22clickhouse    │                0.9 │
│ flushcookies    │                0.9 │
│ clickhousecloud │                0.9 │
│ licheckouts     │                0.9 │
│ clickhousesql   │                0.9 │
│ suchlike        │ 0.8888888888888888 │
│ choleski        │ 0.8888888888888888 │
│ clickhous       │ 0.8888888888888888 │
│ suckhole        │ 0.8888888888888888 │
│ suchlikes       │ 0.8888888888888888 │
│ chuckholes      │ 0.8888888888888888 │
│ suckholes       │ 0.8888888888888888 │
│ clickholes      │ 0.8888888888888888 │
└─────────────────┴────────────────────┘

20 rows in set. Elapsed: 48.896 sec. Processed 37.17 million rows, 12.72 GB (760.21 thousand rows/s., 260.16 MB/s.)
Peak memory usage: 1.83 GiB.
```

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Low performance.
This closes #55078.

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit f5d85732321aa69c201628aaec217b126a8a64be with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/55080/f5d85732321aa69c201628aaec217b126a8a64be/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>✅ success</td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>⏳ pending</td></tr>
<tbody>
</table>

File: src/Functions/FunctionsStringDistance.cpp [hunk 1]
```diff
@@ -185,33 +164,24 @@ struct NameByteHammingDistance
...
-    factory.registerFunction<FunctionByteJaccardIndex>(
-        FunctionDocumentation{.description = R"(Calculates the jaccard similarity index between two bytes strings.)"});
-
     factory.registerFunction<FunctionByteEditDistance>(
-        FunctionDocumentation{.description = R"(Calculates the edit distance between two bytes strings.)"});
-    factory.registerAlias("byteLevenshteinDistance", NameByteEditDistance::name);
+        FunctionDocumentation{.description = R"(Calculates the edit distance between two byte-strings.)"});
+
+    factory.registerAlias("levenshteinDistance", NameEditDistance::name);
 }
```
Comment (ucasfl) [inline src/Functions/FunctionsStringDistance.cpp:186]: The documentation need to be updated as well.

**Comment by antaljanosbenjamin** [comment]
Shouldn't we add a changelog item if we remove some functionality from ClickHouse?

**Comment by alexey-milovidov** [comment]
This is from the unreleased version, so we will only postpone it one or two days before it is added back.

**Comment by alexey-milovidov** [comment]
We are in hurry because we want other functions in the release, but with proper naming, so we don't have to make a backward incompatible change later.

**Pull request merged**